backend/src/controllers/RestaurantController.ts

```diff
@@ -55,24 +58,35 @@ const searchRestaurants = async (req: Request, res: Response) => {
       },
     };
 
+    logger.info(`Found ${total} restaurants for city: ${city}, page: ${page}`);
     res.json(response);
   } catch (error) {
-    console.log(error);
-    res.status(500).json({ message: 'Error searching restaurants' });
+    logger.error(
+      `Error searching restaurants: ${error instanceof Error ? error.message : 'Unknown error'}`,
+      { stack: error instanceof Error ? error.stack : undefined },
+    );
+    next(error);
   }
 };
 
-const getRestaurant = async (req: Request, res: Response) => {
+const getRestaurant = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const restaurantId = req.params.restaurantId;
     const restaurant = await Restaurant.findById(restaurantId).lean();
+
     if (!restaurant) {
+      logger.info(`Restaurant not found: ${restaurantId}`);
       return res.status(404).json({ message: 'Restaurant not found' });
     }
+
+    logger.info(`Fetched restaurant: ${restaurantId}`);
     res.json(restaurant);
   } catch (error) {
-    console.log(error);
-    res.status(500).json({ message: 'Error fetching restaurant' });
+    logger.error(
+      `Error fetching restaurant: ${error instanceof Error ? error.message : 'Unknown error'}`,
+      { stack: error instanceof Error ? error.stack : undefined },
+    );
+    next(error);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2430019945>#d33549</a><span> - 10/22/2024, 07:15 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Consider using a more specific error type instead of the generic `Error`. This could provide more detailed information about the nature of the error, which could be useful for debugging and error handling.
</div>
<div id="fix">
<b>Fix:</b> Create a custom error type for restaurant-related errors and use it instead of the generic `Error`.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,3 +1,7 @@
  import { Request, Response, NextFunction } from 'express';
  import Restaurant from '../models/restaurant';
  import logger from '../utils/logger';
 +
 +class RestaurantError extends Error {}
 +
 +// ... rest of the file ...
 @@ -85,5 +89,5 @@ const getRestaurant = async (req: Request, res: Response, next: NextFunction) => {
      logger.error(
 -      'Error fetching restaurant: ${error instanceof Error ? error.message : 'Unknown error'}',
 -      { stack: error instanceof Error ? error.stack : undefined },
 +      'Error fetching restaurant: ${error instanceof RestaurantError ? error.message : 'Unknown error'}',
 +      { stack: error instanceof RestaurantError ? error.stack : undefined },
      );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged